### PR TITLE
cpu/stm32: fix periph_usb modeling for STM32F1

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

- Fix source selection for STM32F105xx and STM32F107xx
- Fix KConfig modeling for STM32F105xx and STM32F107xx

### Testing procedure

Cannot really be tested yet, as no STM32F105xx and STM32F107xx boards are not yet ported to RIOT. But this is not super complex and scary, so hopefully code review should be sufficient.

### Issues/PRs references

Follow up to https://github.com/RIOT-OS/RIOT/pull/18741

### CODE REVIEW

File: cpu/stm32/periph/Makefile [hunk 1]
```diff
@@ -51,7 +51,7 @@ endif
 
 # Select the correct implementation for `periph_usbdev`
 ifneq (,$(filter periph_usbdev,$(USEMODULE)))
-  ifneq (,$(filter f1 f3 wb,$(CPU_FAM)))
+  ifneq (,$(filter usbdev_synopsys_dwc2,$(USEMODULE)))
```
Comment (benpicco) [inline cpu/stm32/periph/Makefile:54]: Shouldn't that be
```suggestion
  ifeq (,$(filter usbdev_synopsys_dwc2,$(USEMODULE)))
```

**Comment by benpicco** [comment]
Please squash directly

**Review by benpicco — APPROVED** [review]

File: cpu/stm32/periph/Kconfig [hunk 2]
```diff
@@ -13,9 +13,10 @@ config MODULE_PERIPH
     # All STM32 families except for STM32F1, STM32F3 and STM32WG use
     # MODULE_USBDEV_SYNOPSYS_DWC2
     select MODULE_USBDEV_SYNOPSYS_DWC2 if MODULE_PERIPH_USBDEV && !HAS_CPU_STM32WB && !HAS_CPU_STM32F3 && !HAS_CPU_STM32F1
-    # NOTE: In STM32F1 family STM32F105xx and STM32F107xx also use
-    #       MODULE_USBDEV_SYNOPSYS_DWC2. Add those MCUs once the are added to
-    #       KConfig
+    # In STM32F1 family STM32F105xx also uses MODULE_USBDEV_SYNOPSYS_DWC2.
+    select MODULE_USBDEV_SYNOPSYS_DWC2 if MODULE_PERIPH_USBDEV && (CPU_MODEL_STM32F105R8 || CPU_MODEL_STM32F105RB || CPU_MODEL_STM32F105RC || CPU_MODEL_STM32F105V8 || CPU_MODEL_STM32F105VB || CPU_MODEL_STM32F105VC)
```
Comment (gschorcht) [inline cpu/stm32/periph/Kconfig:17]:
```suggestion
    select MODULE_USBDEV_SYNOPSYS_DWC2 if MODULE_PERIPH_USBDEV && CPU_MODEL_STM32F105R8
    select MODULE_USBDEV_SYNOPSYS_DWC2 if MODULE_PERIPH_USBDEV && CPU_MODEL_STM32F105RB
    select MODULE_USBDEV_SYNOPSYS_DWC2 if MODULE_PERIPH_USBDEV && CPU_MODEL_STM32F105RC
    select MODULE_USBDEV_SYNOPSYS_DWC2 if MODULE_PERIPH_USBDEV && CPU_MODEL_STM32F105V8
    select MODULE_USBDEV_SYNOPSYS_DWC2 if MODULE_PERIPH_USBDEV && CPU_MODEL_STM32F105VB
    select MODULE_USBDEV_SYNOPSYS_DWC2 if MODULE_PERIPH_USBDEV && CPU_MODEL_STM32F105VC
```

File: cpu/stm32/periph/Kconfig [hunk 3]
```diff
@@ -13,9 +13,10 @@ config MODULE_PERIPH
...
+    # In STM32F1 family STM32F107xx also uses MODULE_USBDEV_SYNOPSYS_DWC2.
+    select MODULE_USBDEV_SYNOPSYS_DWC2 if MODULE_PERIPH_USBDEV && (CPU_MODEL_STM32F107RB || CPU_MODEL_STM32F107RC || CPU_MODEL_STM32F107VB || CPU_MODEL_STM32F107VC)
```
Comment (gschorcht) [inline cpu/stm32/periph/Kconfig:19]: The same here.

File: cpu/stm32/periph/Kconfig (hunk 2 above)
[the remark this answers is not in the document]
Comment (maribu) [reply cpu/stm32/periph/Kconfig:17]: Agreed, that is easier to read

**Pull request merged**